File: Python Codes/9_Best_Time_to_Buy_and_Sell_Stock_II.py
class Solution:
    def maxProfit(self, prices):
        ans = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i - 1]:
                ans += prices[i] - prices[i - 1]
        return ans

if __name__ == "__main__":
    solution = Solution()
    prices = [7, 1, 5, 3, 6, 4]
    profit = solution.maxProfit(prices)
    print("Maximum Profit:", profit)

File: CPP Codes/9_Best_Time_to_Buy_and_Sell_Stock_II.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class StockProfitCalculator 
{
public:
    int maxProfit(const vector<int>& prices) 
    {
        int ans = 0;
        for (size_t i = 1; i < prices.size(); ++i) 
        {
            ans += max(0, prices[i] - prices[i - 1]);
        }
        return ans;
    }
};

int main() 
{
    StockProfitCalculator calculator;
    vector<int> prices = {7, 1, 5, 3, 6, 4};
    int profit = calculator.maxProfit(prices);
    cout << "Maximum Profit: " << profit << endl;
    return 0;
}
